src/unit_tests/analysisd/test_log.c

```diff
@@ -0,0 +1,244 @@
+    [FIM_DIFF] = {.key = "fim_diff", .value = "diff"},
+    [FIM_ATTRS] = {.key = "fim_attrs", .value = "attributes"},
+    [FIM_ATTRS_BEFORE] = {.key = "fim_attrs_before", .value = "attributes_before"},
+    [FIM_CHFIELDS] = {.key = "fim_chfields", .value = "changed_fields"},
+    [FIM_USER_ID] = {.key = "fim_userid", .value = "2000"},
+    [FIM_USER_NAME] = {.key = "fim_username", .value = "user_name"},
+    [FIM_GROUP_ID] = {.key = "fim_groupid", .value = "2000"},
+    [FIM_GROUP_NAME] = {.key = "fim_groupname", .value = "group_name"},
+    [FIM_PROC_NAME] = {.key = "fim_proc_name", .value = "proc_name"},
+    [FIM_PROC_PNAME] = {.key = "fim_proc_pname", .value = "proc_pname"},
+    [FIM_AUDIT_CWD] = {.key = "fim_audit_cwd", .value = "/audit/cwd"},
+    [FIM_AUDIT_PCWD] = {.key = "fim_audit_pcwd", .value = "/audit/pcwd"},
+    [FIM_AUDIT_ID] = {.key = "fim_audit_id", .value = "6789"},
+    [FIM_AUDIT_NAME] = {.key = "fim_audit_name", .value = "audit_name"},
+    [FIM_EFFECTIVE_UID] = {.key = "fim_effective_uid", .value = "effective_uid"},
+    [FIM_EFFECTIVE_NAME] = {.key = "fim_effective_name", .value = "effective_name"},
+    [FIM_PPID] = {.key = "fim_ppid", .value = "ppid"},
+    [FIM_PROC_ID] = {.key = "fim_proc_id", .value = "proc_id"},
+    [FIM_TAG] = {.key = "fim_tag", .value = "tag1,tag2"},
+    [FIM_SYM_PATH] = {.key = "fim_sym_path", .value = "/sym/path"},
+    [FIM_REGISTRY_ARCH] = {.key = "fim_registry_arch", .value = "[x64]"},
+    [FIM_REGISTRY_VALUE_NAME] = {.key = "fim_registry_value_name", .value = "value_name"},
+    [FIM_REGISTRY_VALUE_TYPE] = {.key = "fim_registry_value_type", .value = "binary"},
+    [FIM_ENTRY_TYPE] = {.key = "fim_entry_type", .value = "registry"},
+    [FIM_EVENT_TYPE_STR] = {.key = "fim_event_type", .value = "modified"}
+};
+
+
+
+
+
+static int test_setup(void **state) {
+    Eventinfo *lf = NULL;
+    os_calloc(1, sizeof(Eventinfo), lf);
+    os_calloc(FIM_NFIELDS, sizeof(DynamicField), lf->fields);
+    os_calloc(2, sizeof(wlabel_t), lf->labels);
+    os_calloc(3, sizeof(char *), lf->last_events);
+    os_calloc(1, sizeof(RuleInfo), lf->generated_rule);
+
+    lf->labels[0].key = "key_label";
+    lf->labels[0].value = "value_label";
+    lf->labels[1].key = NULL;
+    lf->labels[1].value = NULL;
+    lf->nfields = FIM_NFIELDS;
+    lf->last_events[0] = "Last";
+    lf->last_events[1] = "event";
+    lf->last_events[2] = NULL;
+    lf->year = 2000;
+    lf->mon[0] = 'm';
+    lf->mon[1] = '\0';
+    lf->day = 1;
+    lf->hour[0] = 'h';
+    lf->hour[1] = '\0';
+    lf->generated_rule->sigid = 554;
+    lf->generated_rule->level = 123;
+    lf->generated_rule->group = "rule_group";
+    lf->time.tv_sec = 160987966;
+    lf->full_log = "full_log";
+    lf->comment = "comment";
+    lf->hostname = "hostname";
+
+    char aux1[12];
+
+    for (int i = 0; i < lf->nfields; i++) {
+        lf->fields[i].value = df[i].value;
+        lf->fields[i].key = df[i].key;
+    }
```

Comment:
Could we get away with this?
```suggestion
    lf->fields= df;
```